Build broken on Windows by #112

When I build now, I get:
```
   Compiling rustls v0.19.1=======>      ] 19/24: sct, webpki, ring
error[E0432]: unresolved imports `crate::io::rustls_write_vectored_callback`, `crate::io::VectoredCallbackWriter`, `crat
e::io::VectoredWriteCallback`
 --> src\connection.rs:8:5
  |
8 |     rustls_write_vectored_callback, CallbackReader, CallbackWriter, ReadCallback,
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     no `rustls_write_vectored_callback` in `io`
  |     help: a similar name exists in the module: `rustls_write_callback`
9 |     VectoredCallbackWriter, VectoredWriteCallback, WriteCallback,
  |     ^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^ no `VectoredWriteCallback` in `io`
  |     |
  |     no `VectoredCallbackWriter` in `io`
```
What!?

_Originally posted by @gvanem in https://github.com/rustls/rustls-ffi/issues/112#issuecomment-868299417_

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -16,6 +16,14 @@ jobs:
           persist-credentials: false
       - run: make CC=${{ matrix.cc }} PROFILE=release test
 
+  test-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - run: make -f Makefile.Windows
+
   ensure-header-updated:
     runs-on: ubuntu-latest
     steps:

--- a/Makefile.Windows
+++ b/Makefile.Windows
@@ -4,13 +4,13 @@
 #
 export CL=
 
-CRUSTLS_LIB = target/i686-pc-windows-msvc/release/crustls.lib
+CRUSTLS_LIB = target/release/crustls.lib
 
 USE_CLANG_CL ?= 1
 
 CFLAGS     = -nologo -MD -Zi -W3 -O2 -I. -Dssize_t=int -D_CRT_SECURE_NO_WARNINGS
 LDFLAGS    = -nologo -incremental:no
-CARGOFLAGS = --color never --release --target i686-pc-windows-msvc
+CARGOFLAGS = --color never --release
 
 ifeq ($(USE_CLANG_CL),1)
   CC = clang-cl

--- a/README.md
+++ b/README.md
@@ -176,7 +176,3 @@ For this to work, your connection needs to buffer the initial data from the
 client, so these bytes can be replayed to the second connection you use. Do not
 write any data back to the client while your are in the initial connection. The
 client hellos are usually only a few hundred bytes.
-
-
-
-

--- a/src/crustls.h
+++ b/src/crustls.h
@@ -159,8 +159,10 @@ typedef struct rustls_client_config_builder rustls_client_config_builder;
 typedef struct rustls_connection rustls_connection;
 
 /**
- * An alias for `struct iovec` from uio.h. You should cast `const struct rustls_iovec *` to
- * `const struct iovec *`.
+ * An alias for `struct iovec` from uio.h (on Unix) or `WSABUF` on Windows. You should cast
+ * `const struct rustls_iovec *` to `const struct iovec *` on Unix, or `const *LPWSABUF`
+ * on Windows. See [`std::io::IoSlice`] for details on interoperability with platform
+ * specific vectored IO.
  */
 typedef struct rustls_iovec rustls_iovec;
 

--- a/src/io.rs
+++ b/src/io.rs
@@ -100,9 +100,10 @@ impl Write for CallbackWriter {
     }
 }
 
-/// An alias for `struct iovec` from uio.h. You should cast `const struct rustls_iovec *` to
-/// `const struct iovec *`.
-#[cfg(unix)]
+/// An alias for `struct iovec` from uio.h (on Unix) or `WSABUF` on Windows. You should cast
+/// `const struct rustls_iovec *` to `const struct iovec *` on Unix, or `const *LPWSABUF`
+/// on Windows. See [`std::io::IoSlice`] for details on interoperability with platform
+/// specific vectored IO.
 pub struct rustls_iovec {
     _private: [u8; 0],
 }
@@ -119,7 +120,6 @@ pub struct rustls_iovec {
 /// `userdata` is set to the value provided to `rustls_*_session_set_userdata`. In most
 /// cases that should be a struct that contains, at a minimum, a file descriptor.
 /// The buf and out_n pointers are borrowed and should not be retained across calls.
-#[cfg(unix)]
 pub type rustls_write_vectored_callback = Option<
     unsafe extern "C" fn(
         userdata: *mut c_void,
@@ -129,7 +129,6 @@ pub type rustls_write_vectored_callback = Option<
     ) -> rustls_io_result,
 >;
 
-#[cfg(unix)]
 pub(crate) type VectoredWriteCallback = unsafe extern "C" fn(
     userdata: *mut c_void,
     iov: *const rustls_iovec,
@@ -137,13 +136,11 @@ pub(crate) type VectoredWriteCallback = unsafe extern "C" fn(
     out_n: *mut size_t,
 ) -> rustls_io_result;
 
-#[cfg(unix)]
 pub(crate) struct VectoredCallbackWriter {
     pub callback: VectoredWriteCallback,
     pub userdata: *mut c_void,
 }
 
-#[cfg(unix)]
 impl Write for VectoredCallbackWriter {
     fn write(&mut self, buf: &[u8]) -> Result<usize> {
         self.write_vectored(&[IoSlice::new(buf)])
@@ -161,7 +158,7 @@ impl Write for VectoredCallbackWriter {
             cb(
                 self.userdata,
                 // This cast is sound because IoSlice is documented to by ABI-compatible with
-                // iovec on Unix.
+                // iovec on Unix, and with WSABUF on Windows.
                 slices.as_ptr() as *const rustls_iovec,
                 slices.len(),
                 &mut out_n,

